### PR TITLE
Security council mgmt chris misc3

### PR DESCRIPTION
**Pull request by yahgwai**

### CODE REVIEW

File: src/UpgradeExecRouterBuilder.sol [hunk 1]
```diff
@@ -20,38 +24,46 @@ struct ChainAndUpExecLocation {
...
-//                 * chains are arb chains - reachable via inboxes — or L1 (in which case inbox is address(0))
-//                 * can schedule in L1 timelock
 /// @notice Router for target the execution of action contracts in upgrade executors that exist on other chains
-///         Upgrade executors can only be reached by going through a withdrawal and L1 timelock, so this contract
-///         also include these stages when creating/scheduling a route
+///         Routes target an upgrade executor which is either on the host chain, or can be accessed via the inbox
+///         So routes are of two possible forms:
+///         1. Withdrawal => L1Timelock => UpgradeExecutor
+///         2. Withdrawal => L1Timelock => Inbox => UpgradeExecutor
+/// @dev    This contract makes this following:
```
Comment (DZGoldman) [inline src/UpgradeExecRouterBuilder.sol:32]:
```suggestion
/// @dev    This contract makes this following assumptions:
```

File: src/UpgradeExecRouterBuilder.sol [hunk 2]
```diff
@@ -142,38 +159,43 @@ contract UpgradeExecRouterBuilder {
...
             schedTargets,
             schedValues,
             schedData,
-            DEFAULT_PREDECESSOR, // CHRIS: TODO: should we allow others here? could be important? no, use the complex route?
+            predecessor,
             timelockSalt,
             l1TimelockMinDelay
         );
 
+        // withdraw to the l1 timelock
```
Comment (DZGoldman) [inline src/UpgradeExecRouterBuilder.sol:173]:
```suggestion
        // create a message to initiate a withdrawal to the L1 timelock
```

File: src/UpgradeExecRouterBuilder.sol [hunk 3]
```diff
@@ -20,38 +24,46 @@ struct ChainAndUpExecLocation {
...
+///         1. Withdrawal => L1Timelock => UpgradeExecutor
+///         2. Withdrawal => L1Timelock => Inbox => UpgradeExecutor
+/// @dev    This contract makes this following:
+///         * It is deployed on an L2 - more specifically it has access to an ArbSys which allows it to make withdrawal
+///           transactions to a host chain
+///         * It can only target one upgrade executor per chain
+///         * The upgrade executors being targeted are either on the host chain, or are Arbitrum chains reachable
+///           via inboxes on the host chain
+///         * There exists a L1 timelock on the host chain
 contract UpgradeExecRouterBuilder {
```
[the remark this answers is not in the document]
Comment (godzillaba) [reply src/UpgradeExecRouterBuilder.sol:39]: i like route builder as well :)

File: src/UpgradeExecRouterBuilder.sol [hunk 4]
```diff
@@ -142,38 +159,43 @@ contract UpgradeExecRouterBuilder {
...
             address(100),
             abi.encodeWithSelector(ArbSys.sendTxToL1.selector, l1TimelockAddr, timelockCallData)
         );
     }
-    /// @notice creates data for ArbSys for a batch of core governance operations, using common default values for calldata and callvalue
-    /// @param chainIds target chain ids for actions
-    /// @param actionAddresses address of action contracts (on their target chain)
-    /// @param timelockSalt salt for core gov l1 timelock operation
 
+    /// @notice Creates the to address and calldata to be called to execute a route to an action contract
```
Comment (DZGoldman) [inline src/UpgradeExecRouterBuilder.sol:180]:
```suggestion
    /// @notice Creates the to address and calldata to be called to execute a route to a batch of action contracts
```

File: src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol [hunk 5]
```diff
@@ -37,6 +37,7 @@ contract SecurityCouncilNomineeElectionGovernor is
     /// @param owner Owner of the governor (the Arbitrum DAO)
     /// @param quorumNumeratorValue Numerator of the quorum fraction (0.2% = 20)
     /// @param votingPeriod Duration of the voting period (expressed in blocks)
+    ///                     Note that the voting period + nominee vetting duration must be << than 6 months to ensure elections dont overlap
```
Comment (DZGoldman) [inline src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol:40]: remove whitespace?

File: src/security-council-mgmt/governors/modules/SecurityCouncilMemberElectionGovernorCountingUpgradeable.sol [hunk 6]
```diff
@@ -224,36 +224,32 @@ abstract contract SecurityCouncilMemberElectionGovernorCountingUpgradeable is
...
         uint256 decreasingWeightDuration = endBlock - fullWeightVotingDeadline_;
-
         // slope numerator is -votes, slope denominator is decreasingWeightDuration, delta x is blockNumber - fullWeightVotingDeadline_
         // y intercept is votes
         uint256 decreaseAmount =
-            WAD * votes * (blockNumber - fullWeightVotingDeadline_) / decreasingWeightDuration / WAD;
-
-        return decreaseAmount >= votes ? 0 : votes - decreaseAmount;
+            votes * (blockNumber - fullWeightVotingDeadline_) / decreasingWeightDuration;
+        // subtract the decreated amount to get the remaining weight
```
Comment (DZGoldman) [inline src/security-council-mgmt/governors/modules/SecurityCouncilMemberElectionGovernorCountingUpgradeable.sol:251]:
```suggestion
        // subtract the decreased amount to get the remaining weight
```

File: src/UpgradeExecRouterBuilder.sol [hunk 7]
```diff
@@ -142,38 +159,43 @@ contract UpgradeExecRouterBuilder {
...
+    ///         Uses common defaults using for value, calldata and predecessor
+    ///         See Governance Action Contracts for more details
+    /// @param chainIds         Chain ids containing the actions to be called
+    /// @param actionAddresses  Addresses of the action contracts to be called
+    /// @param timelockSalt     A salt for the l1 timelock operation
     function createActionRouteDataWithDefaults(
         uint256[] memory chainIds,
         address[] memory actionAddresses,
-        bytes32 timelockSalt // CHRIS: TODO: can we calculate this in the contract somehow?
+        bytes32 timelockSalt // CHRIS: TODO: should this be a string instead of bytes32? basically this comes down to which is more painful - posting a big description into the tally ui, or posting it seomwhere else, hashing it and then pasting the result into tally?
```
Comment (DZGoldman) [inline src/UpgradeExecRouterBuilder.sol:189]: the description string would need to be pasted regardless, so assuming client-side support the UX should essentially be the same? should prob ask tally

File: src/UpgradeExecRouterBuilder.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (yahgwai) [reply src/UpgradeExecRouterBuilder.sol:39]: Ah yes, that's a typo - will update to route builder

File: src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol (hunk 5 above)
Comment (yahgwai) [reply src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol:40]: I kept it to align with the param above

Comment (yahgwai) [reply src/security-council-mgmt/governors/SecurityCouncilNomineeElectionGovernor.sol:40]: I kept it to align with the param above

File: src/UpgradeExecRouterBuilder.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (yahgwai) [reply src/UpgradeExecRouterBuilder.sol:39]: Renamed to UpgradeExecRouteBuilder everwhere

**Review by godzillaba — APPROVED** [review]

**Review by DZGoldman — APPROVED** [review]

**Pull request merged**